src/Microsoft.AspNet.Routing/Template/TemplateBinder.cs

```diff
@@ -350,7 +373,7 @@ private static bool IsRoutePartNonEmpty(object routePart)
         }
 
         [DebuggerDisplay("{DebuggerToString(),nq}")]
-        private class TemplateBindingContext
+        private struct TemplateBindingContext
```

Comment:
This doesn't have any important state, all the state is part of `_acceptedValues`